firedrake/utility_meshes.py

```diff
@@ -198,6 +198,110 @@ def PeriodicUnitIntervalMesh(ncells):
     return PeriodicIntervalMesh(ncells, length=1.0)
 
 
+def OneElementThickMesh(ncells, Lx, Ly):
+    """
+    Generate a rectangular mesh in the domain with corners [0,0]
+    and [Lx, Ly] with ncells, that is periodic in the x-direction.
+
+    :arg ncells: The number of cells in the mesh.
+    :arg Lx: The width of the domain in the x-direction.
+    :arg Ly: The width of the domain in the y-direction.
+    """
+
+    left = np.arange(ncells, dtype='int32')
+    right = np.roll(left, -1)
+    cells = np.array([left, left, right, right]).T
+    X = np.arange(1.0*ncells)/ncells*Lx
+    Y = 0.*X
+    coords = np.array([X, Y]).T
+
+    # a line of coordinates, with a looped topology
+    plex = PETSc.DMPlex().createFromCellList(2, cells, coords)
```

Comment:
Sorry, I don't understand - what do you want me to do here?

On 15 June 2016 at 09:08, Lawrence Mitchell <notifications@github.com>
wrote:

> In firedrake/utility_meshes.py
> <https://github.com/firedrakeproject/firedrake/pull/803#discussion_r67116830>
> :
>
> > +    and [Lx, Ly] with ncells, that is periodic in the x-direction.
> > +
> > +    :arg ncells: The number of cells in the mesh.
> > +    :arg Lx: The width of the domain in the x-direction.
> > +    :arg Ly: The width of the domain in the y-direction.
> > +    """
> > +
> > +    left = np.arange(ncells, dtype='int32')
> > +    right = np.roll(left, -1)
> > +    cells = np.array([left, left, right, right]).T
> > +    X = np.arange(1.0*ncells)/ncells*Lx
> > +    Y = 0.*X
> > +    coords = np.array([X, Y]).T
> > +
> > +    # a line of coordinates, with a looped topology
> > +    plex = PETSc.DMPlex().createFromCellList(2, cells, coords)
>
> The expand. This creates a DM on all processes, and all processes have the
> cell list and coordinates provided. This DM is the distributed and pain
> ensues. _from_cell_list ensures that only rank 0 makes the DM with the
> provides coords/cells and the other ranks just make data structures of the
> correct shape (but empty).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/firedrakeproject/firedrake/pull/803/files/5c3c02eb550fbc7c577541bc769e35a4ac1b09c3#r67116830>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AF0q8khKzrb-Ylh82I_nCV3at628J8brks5qL7L7gaJpZM4Iw-Dk>
> .
>



-- 
http://www.imperial.ac.uk/people/colin.cotter

www.cambridge.org/9781107663916